internal/services/machinelearning/machine_learning_compute_cluster_resource.go

```diff
@@ -171,21 +170,60 @@ func resourceComputeClusterCreate(d *pluginsdk.ResourceData, meta interface{}) e
 	if err != nil {
 		return err
 	}
-
+	// Get the Machine Learning Workspace...
 	id := machinelearningcomputes.NewComputeID(workspaceID.SubscriptionId, workspaceID.ResourceGroupName, workspaceID.WorkspaceName, d.Get("name").(string))
 
+	workspace, err := mlWorkspacesClient.Get(ctx, *workspaceID)
+	if err != nil {
+		return fmt.Errorf("retrieving %s: %+v", workspaceID, err)
+	}
+
+	workspaceModel := workspace.Model
+	if workspaceModel == nil {
+		return fmt.Errorf("retrieving %s: `model` was nil", workspaceID)
+	}
+
+	if workspaceModel.Sku == nil || workspaceModel.Sku.Tier == nil || workspaceModel.Sku.Name == "" {
+		return fmt.Errorf("retrieving %s: `sku` was nil or empty", workspaceID)
+	}
+
+	if workspaceModel.Location == nil {
+		return fmt.Errorf("retrieving %s: `location` was nil", workspaceID)
+	}
+
+	identity, err := expandIdentity(d.Get("identity").([]interface{}))
+	if err != nil {
+		return fmt.Errorf("expanding `identity`: %+v", err)
+	}
+
 	existing, err := client.ComputeGet(ctx, id)
 	if err != nil {
 		if !response.WasNotFound(existing.HttpResponse) {
 			return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
 		}
 	}
-	if !response.WasNotFound(existing.HttpResponse) {
-		return tf.ImportAsExistsError("azurerm_machine_learning_compute_cluster", id.ID())
```

Comment:
Looks like this might have been removed accidentally? We need to add this back in or the `requiresImport` test fails.